### PR TITLE
Fix deprecation spam

### PR DESCRIPTION
**Pull request by CelticMinstrel**
This fixes the deprecation spam in the log/console by piping it into a brand-new logdomain which by default doesn't output anything at all, effectively adding a new log level below "error".

~The fourth commit is unrelated and would ideally be either discarded or cherry-picked separately, though if the PR is rebased against master it wouldn't really matter.~ It has already been cherry-picked, thanks!

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Also, I may need to make `--log-none` a hidden option for the time being since the man-pages count as translatable and we're in a string freeze? Any thoughts on that?

File: src/scripting/lua_common.cpp [hunk 1]
```diff
@@ -590,12 +590,18 @@ bool luaW_totstring(lua_State *L, int index, t_string &str)
...
 	t_string result;
-	if (!luaW_totstring(L, index, result))
-		luaW_type_error(L, index, "translatable string");
-	return result;
+	if(luaW_totstring(L, index, result)) {
+		return result;
+	}
+	size_t sz;
+	if(const char* cstr = lua_tolstring(L, index, &sz)) {
+		return std::string(cstr, sz);
```
Comment (jyrkive) [inline src/scripting/lua_common.cpp]: Is it okay for `luaW_checktstring()` to just accept regular strings like that?

File: src/version.cpp [hunk 2]
```diff
@@ -193,87 +157,44 @@ namespace {
...
 
 bool operator<(const version_info& l, const version_info& r)
 {
-	return version_numbers_comparison_internal(l, r, LT) || (
-		version_numbers_comparison_internal(l, r, EQUAL) && (
-			(l.special_version().empty() && !r.special_version().empty()) ||
-			(l.special_version() < r.special_version())
-		)
-	);
+	return version_comparison_internal<std::less_equal>(l, r);
```
Comment (jyrkive) [inline src/version.cpp]: Should be `std::less`.

**Comment by jyrkive** [comment]
> Also, I may need to make --log-none a hidden option for the time being since the man-pages count as translatable and we're in a string freeze?

I agree. We're in a string freeze, and therefore the manpage must not change any more.

File: src/scripting/lua_common.cpp (hunk 1 above)
Comment (CelticMinstrel) [reply src/scripting/lua_common.cpp]: I'm not sure, but I don't _think_ it'll hurt anything?

**Comment by CelticMinstrel** [comment]
What about making the default log level for `deprecation` depend on debug mode? For example, entering debug mode automatically increases the severity by one or two levels.

**Comment by jyrkive** [comment]
That would be much better.

**Comment by CelticMinstrel** [comment]
...now I kinda want to know what the heck Codacy said. >_>

**Comment by CelticMinstrel** [comment]
Oh, by the way @jyrkive - it looks like the description strings in the C++ source for each option are not translated, so would it be fine to leave the option not hidden and just omit it from the man-page?

**Comment by jyrkive** [comment]
I'm fine with that.

File: src/scripting/lua_common.cpp (hunk 1 above)
Comment (gfgtdf) [reply src/scripting/lua_common.cpp]: luaW_totstring should already accept them, so i wonder what this does?

Comment (CelticMinstrel) [reply src/scripting/lua_common.cpp]: ...ah, good point. I guess I should just remove this bit then.

**Pull request merged**